backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -67,35 +72,35 @@ public void sendVerificationEmail(String emailAddress) {
       throw new UserException(UserExceptionType.USER_ALREADY_REGISTERED);
     }
 
-    String token = emailUtils.generateJwtEmailToken(emailAddress);
+    String token = JwtUtils.generateEmailToken(emailAddress, JwtUtils.EMAIL_VERIFICATION_PURPOSE);
+
+    Map<String, String> arguments =
+        Map.of("emailAddress", emailAddress, "link", UrlUtil.getEmailVerificationLink(token));
 
-    String confirmationLink = emailUtils.getEmailConfirmationLink() + token;
-    String emailBody = emailUtils.buildVerificationEmail(emailAddress, confirmationLink);
-    Email emailToSend = new Email(emailAddress, SUBJECT_FOR_EMAIL_CONFIRMATION, emailBody, true);
-    emailService.send(emailToSend);
+    emailService.send(new Email(emailAddress, EmailType.EMAIL_VERIFICATION, arguments, true));
   }
 
   @Override
-  public void sendPasswordResetEmail(String email) {
-    if (!userRepository.existsByEmail(email)) {
+  public void sendPasswordResetEmail(String emailAddress) {
+    if (!userRepository.existsByEmail(emailAddress)) {
       throw new UserException(UserExceptionType.USER_NOT_FOUND);
     }
 
-    String token = emailUtils.generateJwtEmailToken(email);
+    String token = JwtUtils.generateEmailToken(emailAddress, JwtUtils.PASSWORD_RESET_PURPOSE);
 
-    String confirmationLink = emailUtils.getPasswordResetLink() + token;
-    String emailBody = emailUtils.buildPasswordResetEmail(email, confirmationLink);
-    Email emailToSend = new Email(email, SUBJECT_FOR_PASSWORD_RESET, emailBody, true);
-    emailService.send(emailToSend);
+    Map<String, String> arguments =
+        Map.of("emailAddress", emailAddress, "link", UrlUtil.getPasswordResetLink(token));
+
+    emailService.send(new Email(emailAddress, EmailType.PASSWORD_RESET, arguments, true));
   }
 
   @Override
-  public String resetPassword(String token, String password) {
-    emailUtils.validateJwtToken(token);
-    String email = emailUtils.getEmailFromJwtEmailToken(token);
+  public String resetPassword(UserPasswordResetRequest userPasswordResetRequest) {
+    String email = JwtUtils.validatePasswordResetToken(userPasswordResetRequest.getToken());
```

Comment:
when validate method is used, usually it is expected a boolean type to be returned. In your case since you return the email value, maybe rename the method to something more descriptive.. **_getValidatedResetPasswordTokenContent_** or something like it

Reply:
Same can and should be applied to every other method from that kind

Reply:
Yeah, current name doesn't indicate that it's returning the email